ARMA: consolidate P and Q

PaulTalbot-INL opened the pull request:
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #784.

##### What are the significant changes in functionality due to this change request?
Changes the `Pmax` and `Pmin` to a single `P` parameter, and similarly for `Q`.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

wangcj05 commented on doc/user_manual/rom.tex:
```diff
@@ -2922,8 +2922,6 @@ \subsubsection{ARMA}
-While plans for optimizing $p$ and $q$ are under consideration, currently setting $Pmin$ and $Pmax$ to the
-same value is required, and similarly for $Qmin$ and $Qmax$.
 %The user is allowed to provide upper and lower limits for $p$
 %and $q$ (see below), and the training process will choose the optimal $p$ and $q$ that fall into the user-specified range.
```
These two lines are not needed, right?

PaulTalbot-INL replied:
It's likely to be added again in the future, but I can remove them for now.

wangcj05 commented on tests/framework/ROM/TimeSeries/ARMA/arma_reseed_precursor.xml:
```diff
@@ -32,10 +32,8 @@
-      <Qmin>1</Qmin>
+      <P>2</P>
+      <Q>1</Q>
       <outTruncation domain="positive">Speed</outTruncation>
```
update the TestInfo regarding these changes

PaulTalbot-INL replied:
done

wangcj05 commented on tests/framework/ROM/TimeSeries/ARMA/arma_reseed_test.xml:
```diff
@@ -1,93 +1,92 @@
-</Simulation>
-
-
+      <revision author="alfoa" date="2018-04-26">Modified location of the test (ARMA folder)</revision>
```
update the TestInfo regarding this modifications.

PaulTalbot-INL replied:
Any changes here were just due to formatting in the conversion script; I think perhaps Windows line endings were removed.  Nothing was revised about the test, as far as I can tell.

wangcj05 commented on tests/framework/ROM/TimeSeries/ARMA/multiple_arma.xml:
```diff
@@ -67,10 +67,8 @@
-      <Qmax>1</Qmax>
-      <Qmin>1</Qmin>
+      <P>2</P>
+      <Q>1</Q>
```
TestInfo should be updated.

PaulTalbot-INL replied:
done

wangcj05 commented on tests/framework/ROM/TimeSeries/ARMA/multiple_fourier.xml:
```diff
@@ -72,16 +71,21 @@
+      <P>3</P>
+      <Q>1</Q>
+      <P>3</P>
+      <Q>1</Q>
```
1. why P and Q and duplicated here
2. TestInfo Should be updated.

PaulTalbot-INL replied:
fixed

wangcj05 commented on tests/framework/ROM/TimeSeries/ARMA/single_fourier.xml:
```diff
@@ -44,13 +44,17 @@
+      <P>2</P>
+      <Q>1</Q>
+      <P>2</P>
+      <Q>1</Q>
```
The same as above?

PaulTalbot-INL replied:
fixed

wangcj05 commented on tests/framework/ROM/TimeSeries/ARMA/test_ARMA.xml:
```diff
@@ -51,13 +52,17 @@
+      <P>2</P>
+      <Q>1</Q>
+      <P>2</P>
+      <Q>1</Q>
```
See above comments

PaulTalbot-INL replied:
fixed

wangcj05 commented on tests/framework/ROM/TimeSeries/ARMA/test_ARMA_parallel.xml:
```diff
@@ -56,13 +56,17 @@
+      <P>2</P>
+      <Q>1</Q>
+      <P>2</P>
+      <Q>1</Q>
```
see above comments

PaulTalbot-INL replied:
fixed

wangcj05 commented on tests/framework/ROM/TimeSeries/ARMA/varma.xml:
```diff
@@ -109,10 +107,8 @@
-      <Qmax>1</Qmax>
-      <Qmin>1</Qmin>
+      <P>3</P>
+      <Q>1</Q>
```
update TestInfo

PaulTalbot-INL replied:
added

wangcj05 commented on tests/framework/ROM/TimeSeries/ARMA/zero_filter.xml:
```diff
@@ -60,18 +60,17 @@
+      <!-- 1 year, 3 months, 1 day -->
       <FourierOrder>1</FourierOrder>
+      <P>3</P>
+      <Q>2</Q>
```
update test info

PaulTalbot-INL replied:
fixed

PaulTalbot-INL commented:
Conversion script has been added.

Draft email:

RAVEN ARMA users:

In recent changes to the ARMA model, the likelihood search for optimal lag parameters P and Q as been removed, and it was required for Pmax and Pmin to be the same value (similar to Q). As of RAVEN PR #785, we have removed Pmax and Pmin and replaced it with simply P, and similar for Q.

A conversion script has been written to help you make changes in your inputs, and is located in raven/scripts/conversionScripts/ARMA_pq.py.  It takes as a command-line argument the input you want to change; for example, "python ~/projects/raven/scripts/conversionScripts/ARMA_pq.py my_input.xml". It will create a backup file, "my_input.bak", for comparison.

For HYBRID users, this change has not been pushed to the RAVEN submodule of HYBRID yet, so no need to take action. We will notify you when this changes.

wangcj05 commented on scripts/conversionScripts/ARMA_pq.py:
```diff
@@ -0,0 +1,53 @@
+
+def convert(tree,fileName=None):
+  """
+    Converts input files to be compatible with merge request #460:
```
update the docstring, should be #785

PaulTalbot-INL replied:
Fixed docstrings, they were left from another conversion script.

wangcj05 commented on scripts/conversionScripts/ARMA_pq.py:
```diff
@@ -0,0 +1,53 @@
+def convert(tree,fileName=None):
+  """
+    Converts input files to be compatible with merge request #460:
+     - Removes "all" node
```
update the explanation regarding this convert, ```all``` is for basic statistical?

wangcj05 commented:
PR looks good to me.

PaulTalbot-INL commented:
It looks like the test timing out is repeatable on other Windows systems as well. Looking into options ...

moosebuild commented:
All [jobs](https://civet.inl.gov/event/37461/) on 0a604de : canceled by @PaulTalbot-INL

Will fail.

PaulTalbot-INL commented:
All tests are passing; the only changes aside from merging in `devel` is to update the segmented ARMA test to use P and Q instead of Pmax and Qmax.

wangcj05 commented:
checklists are satisfied.

wangcj05 commented:
This PR can be merged. @PaulTalbot-INL Could you email the users regarding the changes?

wangcj05 commented:
@PaulTalbot-INL merged.
